ansible/roles/ocp4-workload-cnvlab/tasks/pre_workload.yml

```diff
@@ -0,0 +1,153 @@
+---
+# Implement your Pre Workload deployment tasks here
+# -------------------------------------------------
+
+- name: Install Squid for proxy in lab
+#  package:
+#    name:
+#      - squid
+#      - screen
+#      - qemu-img
+#      - httpd
+#    state: present
+  command: dnf install squid qemu-img httpd -y
```

Comment:
Does `package` (or dnf) not work?

Reply:
It doesn't work -- there's a problem with python3-dnf in the virtualenv python setup... it cannot import the dnf module. So this works around the problem for me.

Reply:
Here's the error:

```
fatal: [bastion.rdo.students.osp.opentlc.com]: FAILED! => {"changed": false, "cmd": "dnf install -y python3-dnf", "msg": "Could not import the dnf python module using /opt/virtualenvs/k8s/bin/python (3.6.8 (default, Dec  5 2019, 15:45:45) [GCC 8.3.1 20191121 (Red Hat 8.3.1-5)]). Please install `python3-dnf` package or ensure you have specified the correct ansible_python_interpreter.", "rc": 0, "results": [], "stderr": "/usr/lib/python3.6/site-packages/dateutil/parser/_parser.py:70: UnicodeWarning: decode() called on unicode string, see https://bugzilla.redhat.com/show_bug.cgi?id=1693751\n  instream = instream.decode()\n\n", "stderr_lines": ["/usr/lib/python3.6/site-packages/dateutil/parser/_parser.py:70: UnicodeWarning: decode() called on unicode string, see https://bugzilla.redhat.com/show_bug.cgi?id=1693751", "  instream = instream.decode()", ""], "stdout": "Updating Subscription Management repositories.\nRed Hat Enterprise Linux 8 for x86_64 - AppStre 4.8 kB/s | 2.8 kB     00:00    \nRed Hat OpenStack Platform 15 Tools for RHEL 8  3.4 kB/s | 2.1 kB     00:00    \nRed Hat Container Native Virtualization 2.3 for 4.1 kB/s | 2.3 kB     00:00    \nRed Hat Ansible Engine 2 for RHEL 8 x86_64 (RPM 3.9 kB/s | 2.3 kB     00:00    \nRed Hat OpenStack Platform 15 for RHEL 8 x86_64 4.0 kB/s | 2.4 kB     00:00    \nRed Hat Enterprise Linux 8 for x86_64 - BaseOS  4.3 kB/s | 2.4 kB     00:00    \nPackage python3-dnf-4.2.17-6.el8.noarch is already installed.\nDependencies resolved.\nNothing to do.\nComplete!\n", "stdout_lines": ["Updating Subscription Management repositories.", "Red Hat Enterprise Linux 8 for x86_64 - AppStre 4.8 kB/s | 2.8 kB     00:00    ", "Red Hat OpenStack Platform 15 Tools for RHEL 8  3.4 kB/s | 2.1 kB     00:00    ", "Red Hat Container Native Virtualization 2.3 for 4.1 kB/s | 2.3 kB     00:00    ", "Red Hat Ansible Engine 2 for RHEL 8 x86_64 (RPM 3.9 kB/s | 2.3 kB     00:00    ", "Red Hat OpenStack Platform 15 for RHEL 8 x86_64 4.0 kB/s | 2.4 kB     00:00    ", "Red Hat Enterprise Linux 8 for x86_64 - BaseOS  4.3 kB/s | 2.4 kB     00:00    ", "Package python3-dnf-4.2.17-6.el8.noarch is already installed.", "Dependencies resolved.", "Nothing to do.", "Complete!"]}
```

Explored on the host:

```
[cloud-user@bastion ~]$ rpm -qa | grep python3-dnf
python3-dnf-4.2.17-6.el8.noarch
python3-dnf-plugin-spacewalk-2.8.5-11.module+el8.1.0+3455+3ddf2832.noarch
python3-dnf-plugins-core-4.0.12-3.el8.noarch

[cloud-user@bastion ~]$ /opt/virtualenvs/k8s/bin/python
Python 3.6.8 (default, Dec  5 2019, 15:45:45) 
[GCC 8.3.1 20191121 (Red Hat 8.3.1-5)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import dnf
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ModuleNotFoundError: No module named 'dnf'
```

Reply:
Now resolved - built my own virtualenv.